sdk/resourcemanager/api-specs.json

```diff
@@ -163,7 +163,8 @@
     "dir": "azure-resourcemanager-monitor",
     "source": "specification/monitor/resource-manager/readme.md",
     "package": "com.azure.resourcemanager.monitor",
-    "args": "--tag=package-2021-04 --add-inner=AutoscaleProfile,ScaleRule,LocalizableString --rename-model=MetricUnit:Unit"
+    "args": "--tag=package-composite-v1 --add-inner=AutoscaleProfile,ScaleRule,LocalizableString,DiagnosticSettingsCategoryResourceCollection --rename-model=MetricUnit:Unit,ActionGroupAutoGenerated:ActivityLogAlertActionGroup,AlertRuleAllOfCondition:ActivityLogAlertAllOfCondition,AlertRuleAnyOfOrLeafCondition:ActivityLogAlertLeafCondition,ActionGroupAutoGenerated:ActivityLogAlertActionGroup",
```

Comment:
why we have these AutoGenerated class? Are them new?

Reply:
the problem for these, are service could potential "fix" them and having a single `ActionGroup` in future.

Reply:
No, they've been renamed in the newer specs.
`ActivityLogAlertActionGroup`'s been rename to `ActionGroup`:

`ActivityLogAlertActionGroup` from `2017-04-01`:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/stable/2017-04-01/activityLogAlerts_API.json#L437

`ActionGroup` from `2020-10-01`:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json#L477

`AutoGenerated` suffix is added by m4, resolving duplicate model name from that in `actionGroups`:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/stable/2022-06-01/actionGroups_API.json#L883

Reply:
wondering if there is minor difference and a few directives could merge them.

or one of them is from a preview that we do not need, and that preview json can be excluded.

Reply:
OK, could we add readme.java.md to do the rename?

Rely on m4 is unreliable, as one does not know which one get to `ActionGroup`, which one to `ActionGroupAutoGenerated`. That might depends on the sequence of json in tag.

Reply:
As the description implies, `ActionGroup` defined in `activityLogAlert` acts as a pointer to that from `actionGroup`:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json#L478

Don't know why they renamed the model...

Reply:
> OK, could we add readme.java.md to do the rename of these duplicated models?
> 
> Rely on m4 is unreliable, as one does not know which one get to `ActionGroup`, which one to `ActionGroupAutoGenerated`. That might depends on the sequence of json in tag.

Sure!